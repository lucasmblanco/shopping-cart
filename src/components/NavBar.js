import React from 'react'
import { Link } from 'react-router-dom';
import { HiHome, HiOutlineShoppingBag, HiPhone } from 'react-icons/hi';

export default function NavBar() {
  return (
    <nav
      data-testid="navigation-bar"
      className="w-40 h-12 rounded-xl rounded-tr-none rounded-br-none flex bg-red-accent shadow-3xl items-center 
    justify-center border-solid border-light-red-accent border-t-2 border-b-2 border-l-2">
      <ul className="flex justify-around w-full ">
        <li className="text-white transition-all duration-500  hover:drop-shadow-nxl ">
          <Link data-testid="home-link" to={'/'} preventScrollReset={true}>
            <HiHome />
          </Link>
        </li>
        <li className="text-white transition-all duration-500 hover:drop-shadow-nxl">
          <Link data-testid="shop-link" to={'/shop'} preventScrollReset={true}>
            <HiOutlineShoppingBag />
          </Link>
        </li>

        <li className="text-white transition-all duration-500 hover:drop-shadow-nxl">
          <Link
            data-testid="contact-link"
            to={'/contact'}
            preventScrollReset={true}>
            <HiPhone />
          </Link>
        </li>
      </ul>
    </nav>
  );
} 