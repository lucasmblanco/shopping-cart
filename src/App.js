import './styles/global.css'
import React, { useEffect, useState } from 'react';
import Header from './components/Header'
import NavBar from './components/NavBar'
import {Routes, Route} from 'react-router-dom'
import Home from './pages/Home'; 
import Shop from './pages/Shop'; 
import Contact from './pages/Contact';
import Cart from './components/Cart'

function App() {

  const [ products, setProducts ] = useState([])
  const [total, setTotal] = useState(0);

  function addQuantity(id, action){
    setProducts(products.map(function(product){
    if (product.id === id && action === 'increment'){
        return {
            ...product, 
            quantity: product.quantity + 1,
           // totalPrice: product.price * (product.quantity + 1)
        }
    } else if(product.id === id && action === 'decrement') {
        return {
            ...product, 
            quantity: product.quantity - 1, 
          //  totalPrice: product.price * (product.quantity - 1)
        }
    } else {
        return {...product }
    }}
    ))
  }

  function deleteProduct(id){
    setProducts(products.filter(product => product.id !== id))
  }

  function addProduct(product){
    setProducts([...products, product])
  }

  function checkProductStatus(productID){
    return products.find(product => product.id === productID)
  }

  useEffect(() => {
    const sumTotal = products.reduce((acc, product) => {
      return  acc + (product.price * product.quantity)
    }, 0);

    setTotal(sumTotal)
  }, [products])


 return (
 <div className="bg-gradient-to-bl from-blue-accent via-lightblue-accent to-lightest-blue-accent animate-gradient-x">
    <Header />
    <div className=' top-5 flex justify-center w-full sticky z-40'>
    <NavBar />
    <Cart products={products} deleteProduct={deleteProduct} addQuantity={addQuantity} total={total}/>
    </div>
    <Routes>
      <Route path='/' element={<Home />} />
      <Route path='/shop' element={<Shop addProduct={addProduct} checkProductStatus={checkProductStatus}/>} />
      <Route path='/contact' element={<Contact />} />
      
    </Routes>

  </div >

 )
}

export default App;
