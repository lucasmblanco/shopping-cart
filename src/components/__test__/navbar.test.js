import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import NavBar from '../NavBar';
import App from '../../App';
import { MemoryRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';

it('should render the component', () => {
  const { asFragment } = render(<NavBar />, { wrapper: MemoryRouter });

  expect(asFragment()).toMatchSnapshot();
});

it('should render three button components', () => {
  render(<NavBar />, { wrapper: MemoryRouter });

  const navigationLinks = screen.getAllByRole('link');

  expect(navigationLinks.length).toEqual(3);
});

it('should render the App with NavBar component', () => {
  render(<App />, { wrapper: MemoryRouter });

  expect(screen.getByTestId('navigation-bar')).toBeInTheDocument();
});

it('should render the chosen component by clicking corresponding link', async () => {
  render(<App />, { wrapper: MemoryRouter });

  const shopLink = screen.getByTestId('shop-link');

  userEvent.click(shopLink);

  expect(await screen.findByTestId('loading-component')).toBeInTheDocument();
});
