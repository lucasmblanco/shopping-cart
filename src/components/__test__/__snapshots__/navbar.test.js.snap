// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the component 1`] = `
<DocumentFragment>
  <nav
    class="w-40 h-12 rounded-xl rounded-tr-none rounded-br-none flex bg-red-accent shadow-3xl items-center 
 justify-center border-solid border-light-red-accent border-t-2 border-b-2 border-l-2"
    data-testid="navigation-bar"
  >
    <ul
      class="flex justify-around w-full "
    >
      <li
        class="text-white transition-all duration-500  hover:drop-shadow-nxl "
      >
        <a
          data-testid="home-link"
          href="/"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10.707 2.293a1 1 0 00-1.414 0l-7 7a1 1 0 001.414 1.414L4 10.414V17a1 1 0 001 1h2a1 1 0 001-1v-2a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 001 1h2a1 1 0 001-1v-6.586l.293.293a1 1 0 001.414-1.414l-7-7z"
            />
          </svg>
        </a>
      </li>
      <li
        class="text-white transition-all duration-500 hover:drop-shadow-nxl"
      >
        <a
          data-testid="shop-link"
          href="/shop"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 11V7a4 4 0 00-8 0v4M5 9h14l1 12H4L5 9z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </a>
      </li>
      <li
        class="text-white transition-all duration-500 hover:drop-shadow-nxl"
      >
        <a
          data-testid="contact-link"
          href="/contact"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2 3a1 1 0 011-1h2.153a1 1 0 01.986.836l.74 4.435a1 1 0 01-.54 1.06l-1.548.773a11.037 11.037 0 006.105 6.105l.774-1.548a1 1 0 011.059-.54l4.435.74a1 1 0 01.836.986V17a1 1 0 01-1 1h-2C7.82 18 2 12.18 2 5V3z"
            />
          </svg>
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
